### PR TITLE
Initialization deprecations

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
#### What? Why?

Fixes some deprecation warnings around class loading during initialization (in preparation for Rails 7):

```
DEPRECATION WARNING: Initialization autoloaded the constant User.

Being able to do this is deprecated. Autoloading during initialization is going
to be an error condition in future versions of Rails.

Reloading does not reboot the application, and therefore code executed during
initialization does not run again. So, if you reload User, for example,
the expected changes won't be reflected in that stale Class object.

This autoloaded constant has been unloaded.

In order to autoload safely at boot time, please wrap your code in a reloader
callback this way:

    Rails.application.reloader.to_prepare do
      # Autoload classes and modules needed at boot time here.
    end

That block runs when the application boots, and every time there is a reload.
For historical reasons, it may run twice, so it has to be idempotent.

Check the "Autoloading and Reloading Constants" guide to learn more about how
Rails autoloads and reloads.
```

#### What should we test?

Green build?

#### Release notes

Improved class loading during app initialization

Changelog Category: Technical changes

### CODE REVIEW

**Review by jibees — APPROVED** [review]

File: config/application.rb [hunk 1]
```diff
@@ -55,21 +55,27 @@ class Application < Rails::Application
     end
 
     initializer "spree.environment", before: :load_config_initializers do |app|
-      app.config.spree = Spree::Core::Environment.new
-      Spree::Config = app.config.spree.preferences # legacy access
+      Rails.application.reloader.to_prepare do
+        app.config.spree = Spree::Core::Environment.new
+        Spree::Config = app.config.spree.preferences # legacy access
```
Comment (mkllnk) [inline config/application.rb:60]: I hope that we can :fire: this soon.

File: config/application.rb [hunk 2]
```diff
@@ -55,21 +55,27 @@ class Application < Rails::Application
...
-      app.config.spree.payment_methods = [
-        Spree::Gateway::Bogus,
-        Spree::Gateway::BogusSimple,
-        Spree::PaymentMethod::Check
-      ]
+      Rails.application.reloader.to_prepare do
+        app.config.spree.payment_methods = [
+          Spree::Gateway::Bogus,
+          Spree::Gateway::BogusSimple,
+          Spree::PaymentMethod::Check
```
Comment (mkllnk) [inline config/application.rb:69]: The modern way is to convert these to strings and reference the classes later in the code.

**Review by mkllnk — APPROVED** [review]
Now it looks like we are using initializers not as they are intended in newer Rails versions. Is there a new practise to configure gems, for example? Should that stuff go in config files?

File: config/application.rb (hunk 2 above)
Comment (Matt-Yorkley) [reply config/application.rb:69]: That sounds good :+1:

**Pull request merged**